addons/stock/models/stock_picking.py

```diff
@@ -1209,6 +1209,7 @@ def action_detailed_operations(self):
                 'show_lots_text': self.show_lots_text,
                 'picking_code': self.picking_type_code,
                 'create': self.state not in ('done', 'cancel'),
+                'show_expiration_date': self.picking_type_id.use_create_lots,
```

Comment:
```suggestion
                'show_expiration_date': self.use_create_lots,
```
The `use_create_lots` field is already there in the picking model.